vaapi/vaapidisplay.cpp

```diff
@@ -320,4 +333,14 @@ DisplayPtr VaapiDisplay::create(const NativeDisplay& display)
 {
     return DisplayCache::getInstance()->createDisplay(display);
 }
+
+DisplayPtr VaapiDisplay::create(const NativeDisplay& display, VAProfile profile)
+{
+    SharedPtr<DisplayCache> cache = DisplayCache::getInstance();
+    const std::string name="hybrid";
```

Comment:
@xuguangxin Some layer has to know that there's a need to use a different driver to use another codec.

The intention with this patch is to push the driver selection as low on the stack as possible.  To be able to do the driver selection at the V4L2 level Yami has to refactor and expose vaSetDriverName and vaInitialize in two different API functions.  I would prefer to do this later as an enhancement.  Also Querying the driver for supported profiles and/or entry points can be done later.